std/uri.d

```diff
@@ -372,6 +372,31 @@ string encodeComponent(Char)(in Char[] uriComponent) if (isSomeChar!Char)
     return URI_Encode(s, URI_Alpha | URI_Digit | URI_Mark);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *
+ * Params:
+ *      values = an associative array containing the values to be encoded.
+ *
+ * Returns:
+ *      A string encoded using www-form-urlencoding.
+ */
+package string urlEncode()(in string[string] values)
+{
+    bool first = true;
+    Appender!string enc;
+    enc.reserve(values.length * 512);
```

Comment:
it depends ... keys are typically small but values easily get 1K+ in machine-oriented APIs